docs/query.md

```diff
@@ -0,0 +1,265 @@
+| `archived:`  | `a:`    | `yes` or `no`          | Filters archived repositories.                             | `archived:yes`                         |
+| `case:`      | `c:`    | `yes`, `no`, or `auto` | Matches case-sensitive or insensitive text.                | `case:yes content:"Foo"`               |
+| `content:`   | `c:`    | Text (string or regex) | Searches content of files.                                 | `content:"search term"`                |
+| `file:`      | `f:`    | Text (string or regex) | Searches file names.                                       | `file:"main.go"`                       |
+| `fork:`      | `f:`    | `yes` or `no`          | Filters forked repositories.                               | `fork:no`                              |
+| `lang:`      | `l:`    | Text                   | Filters by programming language.                           | `lang:python`                          |
+| `public:`    |         | `yes` or `no`          | Filters public repositories.                               | `public:yes`                           |
+| `regex:`     |         | Regex pattern          | Matches content using a regular expression.                | `regex:/foo.*bar/`                     |
+| `repo:`      | `r:`    | Text (string or regex) | Filters repositories by name.                              | `repo:"github.com/user/project"`       |
+| `sym:`       |         | Text                   | Searches for symbol names.                                 | `sym:"MyFunction"`                     |
+| `branch:`    | `b:`    | Text                   | Searches within a specific branch.                         | `branch:main`                          |
+| `type:`      | `t:`    | `filematch`, `filename`, `file`, or `repo` | Limits result types.                   | `type:filematch`                       |
+
+---
+
+### 2. **Negation**
+
+Negate an expression using the `-` symbol.
+
+#### Examples:
+- Exclude a repository:
+  ```plaintext
+  -repo:"github.com/example/repo"
+  ```
+- Exclude a language:
+  ```plaintext
+  -lang:javascript
+  ```
+
+---
+
+### 3. **Grouping**
+
+Group queries using parentheses `()` to create complex logic.
+
+#### Examples:
+- Match either of two repositories:
+  ```plaintext
+  (repo:repo1 or repo:repo2)
+  ```
+- Find test in either python or javascript files:
+  ```plaintext
+  content:test (lang:python or lang:javascript)
+  ```
+
+---
+
+### 4. **Logical Operators**
+
+Use `or` to combine multiple expressions.
+
+#### Examples:
+- Match files in either of two languages:
+  ```plaintext
+  lang:go or lang:java
+  ```
+
+`and` boolean operator is applied automatically when expressions are separated by a space.
+
+---
+
+## Special Query Values
+
+- **Boolean Values**:
+  Use `yes` or `no` for fields like `archived:` or `fork:`.
+
+- **Text Fields**:
+  Text fields (`content:`, `repo:`, etc.) accept:
+  - Strings: `"my text"`
+  - Regular expressions: `/my.*regex/`
```

Comment:
nit: everything is always evaluated as a regular expression, so wrapping things in "/" is not needed

Reply:
ack